core/src/main/javascript/io/Url.js

```diff
@@ -317,12 +321,13 @@
                                     throw this._exception;
                                 }
 
-                                // this allows the stream to return already-parsed JSON
+                                // This allows the stream to return
+                                // already-parsed JSON.
                                 if (result.response.json !== undefined) {
                                     this._json = result.response.json;
                                     this.getJSON = function () { return self._json; };
                                 }
-
+                                
                                 content = result.response.content || textEncoding.getBytes(typeof result.response.body === 'string' ? result.response.body : JSON.stringify(this._json), UTF_8);
```

Comment:
Don't like that we have to stringify this._json - it was already parsed for us. This will increase memory usage on clients that can parse JSON natively.